File: app/src/main/java/com/evaluation/viewmodel/PageViewModel.java
package com.evaluation.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.evaluation.countrylist.CountryQuery;

/**
 * @author Vladyslav Havrylenko
 * @since 16.03.2020
 */
public class PageViewModel extends ViewModel {

    private MutableLiveData<CountryQuery.Country> mLiveData = new MutableLiveData<>();

    public void setResult(CountryQuery.Country results) {
        mLiveData.setValue(results);
    }

    public LiveData<CountryQuery.Country> getResult() {
        return mLiveData;
    }

}

File: app/src/main/java/com/evaluation/viewmodel/PageViewModel.kt
package com.evaluation.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.evaluation.countrylist.CountryQuery.Country

/**
 * @author Vladyslav Havrylenko
 * @since 16.03.2020
 */
class PageViewModel : ViewModel() {

    private var mLiveData = MutableLiveData<Country>()

    fun setResult(results: Country) {
        mLiveData.value = results
    }

    val result: LiveData<Country>
        get() = mLiveData
}